File: app/src/main/java/com/example/octodroid/views/holders/NotificationItemViewHolder.java
package com.example.octodroid.views.holders;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.octodroid.R;
import com.rejasupotaro.octodroid.models.Notification;

import butterknife.Bind;
import butterknife.ButterKnife;

public class NotificationItemViewHolder extends RecyclerView.ViewHolder {
    @Bind(R.id.body)
    TextView bodyTextView;

    public static NotificationItemViewHolder create(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_notification, parent, false);
        return new NotificationItemViewHolder(view);
    }

    public NotificationItemViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(final Notification notification) {
        bodyTextView.setText(notification.toJson());
    }
}

File: app/src/main/kotlin/com/example/octodroid/views/holders/NotificationItemViewHolder.kt
package com.example.octodroid.views.holders

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.bindView
import com.example.octodroid.R
import com.rejasupotaro.octodroid.models.Notification

class NotificationItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
    val bodyTextView: TextView by bindView(R.id.body)

    fun bind(notification: Notification) {
        bodyTextView.text = notification.toJson()
    }

    companion object {

        fun create(parent: ViewGroup): NotificationItemViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.list_item_notification, parent, false)
            return NotificationItemViewHolder(view)
        }
    }
}

